extension/landing-zone/variables.tf

```diff
@@ -0,0 +1,328 @@
+      algorithm         = string
+      protocol          = string
+      health_delay      = number
+      health_retries    = number
+      health_timeout    = number
+      health_type       = string
+      pool_member_port  = string
+      security_group = optional(
+        object({
+          name = string
+          rules = list(
+            object({
+              name      = string
+              direction = string
+              source    = string
+              tcp = optional(
+                object({
+                  port_max = number
+                  port_min = number
+                })
+              )
+              udp = optional(
+                object({
+                  port_max = number
+                  port_min = number
+                })
+              )
+              icmp = optional(
+                object({
+                  type = number
+                  code = number
+                })
+              )
+            })
+          )
+        })
+      )
+    })
+  )
+  default = []
+
+  validation {
+    error_message = "Load balancer names must match the regex pattern ^([a-z]|[a-z][-a-z0-9]*[a-z0-9])$."
+    condition = length(distinct(
+      flatten([
+        # Check through rules
+        for load_balancer in var.load_balancers :
+        # Return false if direction is not valid
+        false if !can(regex("^([a-z]|[a-z][-a-z0-9]*[a-z0-9])$", load_balancer.name))
+      ])
+    )) == 0
+  }
+
+  validation {
+    error_message = "Load Balancer Pool algorithm can only be `round_robin`, `weighted_round_robin`, or `least_connections`."
+    condition = length(
+      flatten([
+        for load_balancer in var.load_balancers :
+        true if !contains(["round_robin", "weighted_round_robin", "least_connections"], load_balancer.algorithm)
+      ])
+    ) == 0
+  }
+
+  validation {
+    error_message = "Incorrect value for the Load Balancer Pool   protocol Valid values are `http`, `https`, or `tcp`."
+    condition = length(
+      flatten([
+        for load_balancer in var.load_balancers :
+        true if !contains(["http", "https", "tcp"], load_balancer.protocol)
+      ])
+    ) == 0
+  }
+
+  validation {
+    error_message = "Pool health delay must be greater than the timeout."
+    condition = length(
+      flatten([
+        for load_balancer in var.load_balancers :
+        true if load_balancer.health_delay < load_balancer.health_timeout
+      ])
+    ) == 0
+  }
+
+  validation {
+    error_message = "Load Balancer Pool Health Check Type can only be `http`, `https`, or `tcp`."
+    condition = length(
+      flatten([
+        for load_balancer in var.load_balancers :
+        true if !contains(["http", "https", "tcp"], load_balancer.health_type)
+      ])
+    ) == 0
+  }
+
+  validation {
+    error_message = "Each load balancer must have a unique name."
+    condition     = length(distinct(var.load_balancers[*].name)) == length(var.load_balancers[*].name)
+  }
+}
```

Comment:
I don't think we need to duplicate any validation logic here. Its already in the root level module which this extension will call, so no need to duplicate and maintain here too